pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/builder/DefaultRealtimeRoutingTableBuilder.java

```diff
@@ -68,7 +69,7 @@ public void computeOnExternalViewChange(String tableName, ExternalView externalV
   }
 
   @Override
-  public Map<String, List<String>> getRoutingTable(RoutingTableLookupRequest request) {
+  public Map<String, List<String>> getRoutingTable(RoutingTableLookupRequest request, SegmentSelector segmentSelector) {
```

Comment:
Should selector be passed into this api, or should it belong inside of the builder(s)?

Reply:
As discussed offline, we concluded that broker routing builder requires the re-structuring on class hierarchy. I will add the comments about it and will address later with the separate clean-up pr.